exe/3scale-update

```diff
@@ -0,0 +1,270 @@
+
+    source_hits = source_metrics.find{ |metric| metric['system_name'] == 'hits' } or raise 'missing hits metric'
+    target_hits = target_metrics.find{ |metric| metric['system_name'] == 'hits' } or raise 'missing hits metric'
+
+    source_methods = source_client.list_methods(source_service_id, source_hits['id'])
+    target_methods = target_client.list_methods(target_service_id, target_hits['id'])
+
+    puts "source service hits metric #{source_hits['id']} has #{source_methods.size} methods"
+    puts "target service hits metric #{target_hits['id']} has #{target_methods.size} methods"
+
+    missing_methods = source_methods.reject { |source_method|  target_methods.find{|target_method| compare_hashes(source_method, target_method, ['system_name']) } }
+
+    puts "creating #{missing_methods.size} missing methods on target service..."
+    missing_methods.each do |method|
+      target = { friendly_name: method['friendly_name'], system_name: method['system_name'] }
+      target_client.create_method(target_service_id, target_hits['id'], target)
+    end
+
+    target_metrics = target_client.list_metrics(target_service_id)
+
+    puts "source service has #{source_metrics.size} metrics"
+    puts "target service has #{target_metrics.size} metrics"
+
+    missing_metrics = source_metrics.reject { |source_metric| target_metrics.find{|target_metric| compare_hashes(source_metric, target_metric, ['system_name']) } }
+
+    missing_metrics.map do |metric|
+      metric.delete('links')
+      target_client.create_metric(target_service_id, metric)
+    end
+
+    puts "created #{missing_metrics.size} metrics on the target service"
+  end
+
+  def copy_application_plans
+    source_plans = source_client.list_service_application_plans(source_service_id)
+    target_plans = target_client.list_service_application_plans(target_service_id)
+
+    puts "source service has #{source_plans.size} application plans"
+    puts "target service has #{target_plans.size} application plans"
+
+    missing_application_plans = source_plans.reject { |source_plan| target_plans.find{|target_plan| source_plan.fetch('system_name') == target_plan.fetch('system_name') } }
+
+    puts "creating #{missing_application_plans.size} missing application plans..."
+
+    missing_application_plans.each do |plan|
+      plan.delete('links')
+      plan.delete('default') # TODO: handle default plans
+
+      if plan.delete('custom') # TODO: what to do with custom plans?
+        puts "skipping custom plan #{plan}"
+      else
+        target_client.create_application_plan(target_service_id, plan)
+      end
+    end
+
+    puts "updating limits for application plans..."
+
+    application_plan_mapping = target_client.list_service_application_plans(target_service_id).map do |plan_target|
+      plan = source_plans.find{|plan| plan.fetch('system_name') == plan_target.fetch('system_name') }
+
+      [plan['id'], plan_target['id']]
+    end
```

Comment:
I think this does not work. It does not call `.to_h` so it is just array of tuples and then the next each expects it to be hash